File: leetcode/283-moveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int point = 0, len = nums.length;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] != 0)
                nums[point++] = nums[i];
        }
        for( ; point < len; point++)
            nums[point] = 0;
    }
}

File: leetcode/283-moveZeroes.c
void moveZeroes(int* nums, int numsSize) {
    int p = 0;
    for(int i = 0; i < numsSize; i++){
        if(*(nums+i) != 0){
            *(nums+p) = *(nums+i);
            p++;
        }
    }
    while(p < numsSize){
        *(nums+p) = 0;
        p++;
    }
}
